src/logic/editor_game_base.cc

```diff
@@ -514,7 +514,6 @@ Immovable& EditorGameBase::do_create_immovable(const Coords& c,
 	const ImmovableDescr& descr =
 	   *(type == MapObjectDescr::OwnerType::kTribe ? tribes().get_immovable_descr(idx) :
 	                                                 world().get_immovable_descr(idx));
-	assert(&descr);
```

Comment:
Did you remove this line by accident?

Reply:
My compiler was always annoying me with "The compiler can guarantee that the address of descr will never be null". Same for the other removed assert.

Reply:
Ah, probably because we're fetching a reference instead of a pointer.

As a substitute, we could assert or check for tribes()/world.immovable.exists() before fetching the descr.

Reply:
How about refactoring the `get_xxx_descr()` functions to return a reference or throw a WException? Not in this branch of course.

Reply:
Sounds good. Let's wait for the tribes_filtering too, because the changes will be all over the code base.